### PR TITLE
Fix: recursive sync

### PR DESCRIPTION
**Pull request by 3lvis**
The root organization has a `parentUnit = nil` that isn't taken into account when filtering the changes
Sync should check if the entity been synced has a parent entity and add it to the predicate:
Predicate: `parentUnit = nil` for root objects 
otherwise it will fetch all organizations, even the ones with parents
I pushed a silly fix

### CODE REVIEW

File: Source/Sync.m [hunk 1]
```diff
@@ -49,7 +49,7 @@ + (void)changes:(NSArray *)changes
         NSError *error = nil;
         NSManagedObject *safeParent = [parent sync_copyInContext:backgroundContext
                                                            error:&error];
-        NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K = %@", [parent.entity.name lowercaseString], safeParent];
+        NSPredicate *predicate = [NSPredicate predicateWithFormat:@"parentUnit = %@", [parent.entity.name lowercaseString], safeParent];
```
Comment (kostiakoval) [inline Source/Sync.m]: hardcoded `parentUnit` in Sync ?

Comment (3lvis) [inline Source/Sync.m]: @kostiakoval Just to prove that this fixes the issue, this isn't the real solution.

File: Source/Sync.m [hunk 2]
```diff
@@ -77,6 +78,15 @@ + (void)changes:(NSArray *)changes
     NSString *remoteKey = [entity sync_remoteKey];
     NSParameterAssert(remoteKey);
 
+    if (!parent && !predicate) {
+        NSEntityDescription *current = [NSEntityDescription entityForName:entityName
```
Comment (kostiakoval) [inline Source/Sync.m]: can you use `entity`variable instead of making `current`here. It's the same right?

Comment (kostiakoval) [inline Source/Sync.m]: It took me some time to understand this code. maybe it would be smart to add some comment or making a method for it ?

**Comment by 3lvis** [comment]
Thanks @kostiakoval!

**Pull request merged**